client/web/web.go

```diff
@@ -756,6 +732,53 @@ func (s *Server) tailscaleUp(ctx context.Context, st *ipnstate.Status, postData
 	}
 }
 
+type tailscaleUpOptions struct {
+	Reauthenticate bool
+	ForceLogout    bool
+}
+
+// serveTailscaleUp serves requests to /api/up.
+// If the user needs to authenticate, an authURL is provided in the response.
+func (s *Server) serveTailscaleUp(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	st, err := s.lc.Status(r.Context())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	var opt tailscaleUpOptions
+	type mi map[string]any
+	if err := json.NewDecoder(r.Body).Decode(&opt); err != nil {
+		w.WriteHeader(400)
+		json.NewEncoder(w).Encode(mi{"error": err.Error()})
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	var reauth, logout bool
+	if opt.Reauthenticate {
+		reauth = true
+	}
+	if opt.ForceLogout {
+		logout = true
+	}
+	s.logf("tailscaleUp(reauth=%v, logout=%v) ...", reauth, logout)
```

Comment:
Could just use opt.Reauthenticate and opt.ForceLogout here directly 🤷🏽‍♀️

Reply:
done